cumulus/primitives/storage-weight-reclaim/src/lib.rs

```diff
@@ -183,13 +183,13 @@ where
 			if consumed_weight > benchmarked_weight {
 				log::error!(
 					target: LOG_TARGET,
-					"Benchmarked storage weight smaller than consumed storage weight. benchmarked: {benchmarked_weight} consumed: {consumed_weight} unspent: {unspent}"
+					"Benchmarked storage weight smaller than consumed storage weight. call: {call:?} benchmarked: {benchmarked_weight} consumed: {consumed_weight} unspent: {unspent}"
```

Comment:
Instead of the call we could print the `frame_system::Pallet::extrinsic_count()`. Then you have the index of the transaction.

Reply:
I find that less straightforward. You might check the logs at a later point in time, when the state could, for instance, have been pruned. In that scenario you wouldn't be able to retrieve the block (after taking the time to inspect the block in PolkadotJS).

Also, it'd take more time to figure out which extrinsic is. My purpose is to increase as much as possible developer productivity, and the most go-to way I found is this one.

That being said, I understand passing the call and printing its debug information has a performance penalty. Still, I think it'd be better off with the proposed changes, as a small performance penalty could potentially save hours of debugging and improve overall productivity.

Still... if you think otherwise, I'll change it as per your suggestion. But I'd prefer to softly push for the proposed solution, as in my opinion it represents the best trade-off.

Reply:
> You might check the logs at a later point in time, when the state could, for instance, have been pruned. In that scenario you wouldn't be able to retrieve the block (after taking the time to inspect the block in PolkadotJS).

State pruning is different to blocks pruning. Block pruning needs to be enabled directly.

We don't have that much memory and wasting the memory just for some debugging which is only required in 0.01% of the cases is not a good idea.

> It does access an storage item. Wouldn't that introduce a higher performance penalty than the proposed solution?

The storage item will only be accessed when the log level is enabled . For the release build on chain logging should be disabled any way. Thus, this should not lead to any performance impacts on production networks.

Reply:
> some debugging which is only required in 0.01% of the cases is not a good idea.

I'd say an incorrectly benchmarked custom pallet in a parachain is much more common than that. Still, I implemented it as per your suggestion.

Reply:
Just take all the nodes into account that need to execute the block and that are not used for debugging. Debugging is a small fraction and you get the index of the transaction and that gives you all the information for free.

Reply:
@bkchr please double check, but to me it looks like `ExtrinsicCount` is not updated on the fly, but only at the end of the block here (also in line with the docs): https://github.com/paritytech/polkadot-sdk/blob/fcea4ff4a9f7f334fb488c15c2be3f26061c3bf8/substrate/frame/system/src/lib.rs#L2048-L2054

I think we should use `extrinsic_index()` instead which seems to be updated after every xt. https://github.com/paritytech/polkadot-sdk/blob/fcea4ff4a9f7f334fb488c15c2be3f26061c3bf8/substrate/frame/system/src/lib.rs#L2042-L2042

Reply:
:see_no_evil: